provisioning/service/tests/Config/TpmAttestationTests.cs

```diff
@@ -0,0 +1,219 @@
+        }
+
+        [TestMethod]
+        [TestCategory("DevService")]
+        public void TpmAttestation_ThrowsOnEnptyStorageRootKey()
+        {
+            // arrange
+            string endorsementKey = KEY;
+            string storageRootKey = "";
+
+            // act - assert
+            TestAssert.Throws<ArgumentException>(() => new TpmAttestation(endorsementKey, storageRootKey));
+        }
+
+        [TestMethod]
+        [TestCategory("DevService")]
+        public void TpmAttestation_ThrowsOnNonBase64StorageRootKey()
+        {
+            // arrange
+            string endorsementKey = KEY;
+            string storageRootKey = "This is not base64";
+
+            // act - assert
+            TestAssert.Throws<ArgumentException>(() => new TpmAttestation(endorsementKey, storageRootKey));
+        }
+
+        /* SRS_TPM_ATTESTATION_21_003: [The constructor shall store the provided endorsementKey and storageRootKey.] */
+        [TestMethod]
+        [TestCategory("DevService")]
+        public void TpmAttestation_SucceedOnValidEndorsementKey()
+        {
+            // arrange
+            string endorsementKey = KEY;
+
+            // act
+            TpmAttestation tpmAttestation = new TpmAttestation(endorsementKey);
+
+            // assert
+            Assert.AreEqual(endorsementKey, tpmAttestation.EndorsementKey);
+            Assert.IsNull(tpmAttestation.StorageRootKey);
+        }
+
+        [TestMethod]
+        [TestCategory("DevService")]
+        public void TpmAttestation_SucceedOnValidEndorsementKeyAndStorageRootKey()
+        {
+            // arrange
+            string endorsementKey = KEY;
+            string storageRootKey = KEY;
+
+            // act
+            TpmAttestation tpmAttestation = new TpmAttestation(endorsementKey, storageRootKey);
+
+            // assert
+            Assert.AreEqual(endorsementKey, tpmAttestation.EndorsementKey);
+            Assert.AreEqual(storageRootKey, tpmAttestation.StorageRootKey);
+        }
+
+        /* SRS_TPM_ATTESTATION_21_004: [The TpmAttestation shall provide means to serialization and deserialization.] */
+        [TestMethod]
+        [TestCategory("DevService")]
+        public void TpmAttestation_SucceedOnSerialization()
+        {
+            // arrange
+            string endorsementKey = KEY;
+            string storageRootKey = KEY;
+            string expectedJson = 
+                "{" +
+                "  \"endorsementKey\":\""+endorsementKey+"\"," +
+                "  \"storageRootKey\":\"" + storageRootKey + "\"" +
+                "}";
+            TpmAttestation tpmAttestation = new TpmAttestation(endorsementKey, storageRootKey);
+
+            // act
+            string json = Newtonsoft.Json.JsonConvert.SerializeObject(tpmAttestation);
+
+            // assert
+            TestAssert.assertJson(expectedJson, json);
+        }
+
+        [TestMethod]
+        [TestCategory("DevService")]
```

Comment:
What is the purpose of this TestCategory?

Reply:
I'm not sure too.

Reply:
I'll keep it for now